pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -428,6 +438,25 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
     }
 
+    logBrokerResponse(requestStatistics, requestId, query, compilationStartTimeNs, brokerRequest,
+        numUnavailableSegments, serverStats, brokerResponse, executionEndTimeNs);
+    return brokerResponse;
+  }
+
+  /**
+   * Given a {@link BrokerRequest}, this function will determine if we can return a response without server-side query
+   * evaluation. This happens when the optimizer determines that the entire WHERE clause evaluates to false.
+   */
+  private boolean isResponsePossible(BrokerRequest brokerRequest) {
+    return brokerRequest == null || brokerRequest.getPinotQuery() == null || (
```

Comment:
(Critical) `brokerRequest.getPinotQuery()` will always be `null` for PQL query

Reply:
Added check for making sure we are doing this only for SQL queries.